packages/manager/modules/bm-server-components/src/general-information/inputs/controller.js

```diff
@@ -0,0 +1,75 @@
+import findLastIndex from 'lodash/findLastIndex';
+import remove from 'lodash/remove';
+
+export default class BmServerComponentsInputsController {
+  /* @ngInject */
+  constructor($q, $scope, $stateParams, $translate) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    angular.forEach(this.installation.inputs, (value, key) => {
+      if (value.type === 'enum' && !value.mandatory) {
+        this.installation.inputs[key].enum = [''].concat(value.enum);
+        // you have an optional select dropdown so you want to be able to unset to none
+      }
+    });
+
+    // inputs
+    // let inputsIndex = 0;
+    this.installation.input = {};
+    angular.forEach(this.installation.inputs, (inputItem) => {
+      if (inputItem.type === 'keyValue') {
+        this.installation.input[inputItem.name] = [];
+      } else if (inputItem.type === 'number') {
+        this.installation.input[inputItem.name] = +inputItem.default;
+        this.installation.input[inputItem.name] =
+          inputItem.default === null ? 0 : +inputItem.default;
+      } else if (inputItem.type === 'boolean') {
+        this.installation.input[inputItem.name] =
+          inputItem.default !== null && inputItem.default === 'true';
+      } else {
+        this.installation.input[inputItem.name] =
+          inputItem.default === null ? '' : inputItem.default?.toString();
+      }
+    });
+  }
+
+  keyValueInit = function keyValueInit(name, mode) {
+    if (this.installation.input[name].length !== 0) {
+      return '';
+    }
+    const indexItem = findLastIndex(this.installation.inputs, {
+      name,
+    });
+    if (
+      indexItem !== -1 &&
+      this.installation.inputs[indexItem].default &&
+      this.installation.inputs[indexItem].default.replace(', ', ',').split(',')
+        .length === 2
+    ) {
+      return this.installation.inputs[indexItem].default
+        .replace(', ', ',')
+        .split(',')[mode === 'value' ? 1 : 0];
+    }
+    return '';
```

Comment:
```suggestion
    let keyValueInit = '';
    if (this.installation.input[name].length == 0) {
      const indexItem = findLastIndex(this.installation.inputs, {
        name,
      });
      const inputItem = this.installation.inputs[indexItem];

      if (inputItem?.default) {
        const splitDefaults = inputItem.default.split(/, ?/);
        
        if (splitDefaults.length === 2) {
          keyValueInit = splitDefaults[mode === 'value' ? 1 : 0];
        }
      }
    }

    return keyValueInit;
```

Reply:
What if findLastIndex returns -1 (no found) ?
@dectotam can you re-review this part of the new updated version ?

Reply:
If lastIndex is -1, then inputItem will be undefined, so the condition inputItem?.default will be false, and keyValueInit will be returned as default empty value ''.

Reply:
I just disagree with the `let keyValueInit = '';` and single `return keyValueInit` because it makes too many nested if compared to a first
```js
if (this.installation.input[name].length === 0) {
    return '';
}
```
then we save an if indentation `if (this.installation.input[name].length !== 0) {` and the nested if conditions that we  need to put inside

Reply:
@dectotam check my change, it's not exactly your suggestion because of my previous comment and @JacquesLarique's suggestion regarding `findLastIndex` function's deprecated `lodash` depedency